### PR TITLE
Added new Hexagon shape in the Network

### LINKED ISSUE #3418: can we add hexagon as shape for node in network
I am using the **Network** layout and I wanted to show node in **hexagon shape**, how can I do this, can we use SVG script in vis js

### PR DESCRIPTION
**Pull request by rg2609**
Fixed #3418 and added Hexagon shape for nodes.

### CODE REVIEW

**Comment by wimrijnders** [comment]
From the Travis build:

```
[07:50:48] Using gulpfile ~/build/almende/vis/gulpfile.js
[07:50:48] Starting 'lint'...
[07:50:53] 

/home/travis/build/almende/vis/lib/network/modules/components/nodes/shapes/Hexagon.js
   5:1   error  Missing JSDoc comment  require-jsdoc
   6:14  error  Missing JSDoc comment  require-jsdoc
  10:7   error  Missing JSDoc comment  require-jsdoc
  14:19  error  Missing JSDoc comment  require-jsdoc
```

Please fix all error messages from `gulp lint`.

File: lib/network/shapes.js [hunk 1]
```diff
@@ -320,5 +320,22 @@ if (typeof CanvasRenderingContext2D !== 'undefined') {
...
 
+  /**
+   * Draw a Hexagon shape with 6 sides
+   * @param {Number} x horizontal center
+   * @param {Number} y vertical center
+   * @param {Number} r   radius
+   */
+  CanvasRenderingContext2D.prototype.hexagon = function (x, y, r) {
+    this.beginPath();
+    var sides = 6
```
Comment (wimrijnders) [inline lib/network/shapes.js]: ;

File: lib/network/shapes.js [hunk 2]
```diff
@@ -320,5 +320,22 @@ if (typeof CanvasRenderingContext2D !== 'undefined') {
...
+    var a = (Math.PI*2)/sides;
+    this.moveTo(x+r,y);
+    for (var i = 1; i < sides; i++) {
+      this.lineTo(x+r*Math.cos(a*i),y+r*Math.sin(a*i));
+    }
+      this.closePath();
```
Comment (wimrijnders) [inline lib/network/shapes.js]: Indent

**Comment by wimrijnders** [comment]
Can you add `hexagon` to the documentation as well please?

In file `docs/network/nodes.html`, there are three lists of node shapes, for properties:
- `scaling`
- `shape`
- `size`

Please add 'hexagon' to this list - anywhere you see `square`, `hexagon` must also be present.

File: lib/network/shapes.js (hunk 2 above)
Comment (rg2609) [reply lib/network/shapes.js]: Done

File: lib/network/shapes.js (hunk 1 above)
Comment (rg2609) [reply lib/network/shapes.js]: Done

**Comment by wimrijnders** [comment]
Travis checks are failing, but this is not your fault. It's a known issue (and it's driving me crazy).

If you want to do something about this, please add the following:

In file `test/Network.test.js` changed function `after()` to:

```js
  after(function() {
    try {
      this.jsdom_global();
    } catch(e) {
      if (e.message() === 'window is undefined') {
        console.warning("'" + e.message() + "' happened again");
      } else {
        throw e;
      }
    }
  });
```

This is *totally* unrelated to your changes. But it would be a good test to see if it helps Travis.

**Comment by wimrijnders** [comment]
😞 Didn't help. But thanks for doing it anyway. Your PR is good, despite the failing test. Nothing to do with the PR.

**Comment by macleodbroad-wf** [comment]
lgtm +1

**Comment by rg2609** [comment]
How much time will it take to merge this PR?

**Comment by wimrijnders** [comment]
Not too much. Someone else will do it and it depends on his availability. I think it will happen this weekend.

**Pull request merged**